### PR TITLE
@task PDA-1241 GET /items/{itemId} - build graphQL query

### PR DESCRIPTION
**Pull request by tren-parasoft**
GET /items/{itemId} - build graphQL query

### CODE REVIEW

File: src/test/java/com/parasoft/demoapp/graphql/ItemGraphQLDataFetcherTest.java [hunk 1]
```diff
@@ -106,6 +118,42 @@ public void test_getItems_incorrectAuthentication() throws IOException {
...
+        assertThat(graphQLResponse).isNotNull();
+        assertThat(graphQLResponse.isOk()).isTrue();
+        graphQLResponse.assertThatNoErrorsArePresent()
+                .assertThatDataField().isNotNull()
+                .and()
+                .assertThatField(GET_ITEM_BY_ITEM_ID_DATA_JSON_PATH)
+                .as(ItemEntity.class)
+                .hasFieldOrPropertyWithValue("id", 1L);
+    }
+
```
Comment (rwang-parasoft) [inline src/test/java/com/parasoft/demoapp/graphql/ItemGraphQLDataFetcherTest.java:140]: Add a test to test with "incorrectAuthentication".

File: src/test/java/com/parasoft/demoapp/graphql/ItemGraphQLDataFetcherTest.java [hunk 2]
```diff
@@ -106,6 +118,42 @@ public void test_getItems_incorrectAuthentication() throws IOException {
...
+    @Test
+    public void getItemByItemId_NotFound() throws Throwable {
```
Comment (rwang-parasoft) [inline src/test/java/com/parasoft/demoapp/graphql/ItemGraphQLDataFetcherTest.java:142]: A small change to keep the same style with other test names - change the name to `getItemByItemId_notFound`.

**Review by rwang-parasoft — APPROVED** [review]

File: src/main/resources/static/schema.graphql [hunk 3]
```diff
@@ -11,6 +11,7 @@ type Query {
     getOrderByOrderNumber(orderNumber: String): Order
     #item
     getItems(categoryId: Int, regions: [RegionType], searchString: String, page: Int, size: Int, sort: [String]): ItemPageInfo
+    getItemByItemId(itemId: ID): Item
```
Comment (yshi-parasoft) [inline src/main/resources/static/schema.graphql:14]: Please change "itemId" field to be required to be consistent with corresponding REST API.

File: src/test/resources/graphql/items/getItem.graphql [hunk 4]
```diff
@@ -0,0 +1,12 @@
+query GetItemByItemId($itemId: ID) {
```
Comment (yshi-parasoft) [inline src/test/resources/graphql/items/getItem.graphql:1]: Please update the field to be required as "!ID"

File: src/main/java/com/parasoft/demoapp/graphql/ItemGraphQLDataFetcher.java [hunk 5]
```diff
@@ -82,4 +80,24 @@ public DataFetcher<PageInfo<ItemEntity>> getItems() {
...
         };
     }
+
+    public DataFetcher<ItemEntity> getItemByItemId() {
+        return dataFetchingEnvironment -> {
+            try {
+                Map<String, String> uriVariables = new HashMap<>();
+                if (dataFetchingEnvironment.containsArgument("itemId")) {
+                    uriVariables.put("itemId", dataFetchingEnvironment.getArgument("itemId"));
+                }
```
Comment (yshi-parasoft) [inline src/main/java/com/parasoft/demoapp/graphql/ItemGraphQLDataFetcher.java:90]: Please change this block to be:
<pre>String itemId= dataFetchingEnvironment.getArgument("itemId");
if (itemId!= null && !itemId.isEmpty()) {
    uriVariables.put("itemName", itemId);
}</pre>

**Pull request merged**